libbeat/monitoring/report/elasticsearch/elasticsearch.go

```diff
@@ -386,18 +388,35 @@ func makeMeta(beat beat.Info) common.MapStr {
 	}
 }
 
-func getClusterUUID() string {
+func getClusterUUID() (string, bool) {
+	const esOutputName = "elasticsearch"
+
 	stateRegistry := monitoring.GetNamespace("state").GetRegistry()
+	outputRegistry := stateRegistry.GetRegistry("output")
+	if outputRegistry == nil {
+		return "", false
+	}
+
+	snapshot := monitoring.CollectFlatSnapshot(outputRegistry, monitoring.Full, false)
+	outputName := snapshot.Strings["name"]
+	if outputName != esOutputName {
+		// Output is not Elasticsearch, we don't expect to get a cluster UUID from this output
+		return "", true
+	}
+
 	outputsRegistry := stateRegistry.GetRegistry("outputs")
 	if outputsRegistry == nil {
-		return ""
+		return "", true
 	}
 
 	elasticsearchRegistry := outputsRegistry.GetRegistry("elasticsearch")
 	if elasticsearchRegistry == nil {
-		return ""
+		return "", true
 	}
 
-	snapshot := monitoring.CollectFlatSnapshot(elasticsearchRegistry, monitoring.Full, false)
-	return snapshot.Strings["cluster_uuid"]
+	snapshot = monitoring.CollectFlatSnapshot(elasticsearchRegistry, monitoring.Full, false)
+	clusterUUID := snapshot.Strings["cluster_uuid"]
+
+	okToPublish := outputName == esOutputName && clusterUUID != ""
+	return clusterUUID, okToPublish
```

Comment:
I know we don't have a better solution for this as of know, but all this code poking around in globals + dynamic data makes me uneasy. There is a reason `monitoring` has no real getters, and creating snapshots is quite some hack around this.

As monitoring is internal, maybe we can hook it up with the outputs. If there is an output that has an idea about ClusterUUID, it will expose a method for this (a simple typecast would help). Then this code would become agnostic to the actual output, allowing us to implement some magic like:
- for elasticsearch output to connect and query cluster UUID
- logstash output with new lumberjack protocol (we plan to have some RPC mechanism) could collect UUID right from the logstash endpoint.

e.g. 

```
func getClusterUUID(output ...) (string, bool) {
  if clusterAware, ok := output.(interface{ ClusterUUID() (string, error) }); ok {
    id, err := clusterAware.ClusterUUID()
    return id, err == nil
  }
  return "", false
}
```

Reply:
Yup, doing this lookup via a type assertion on the output is much better. Will change!

Reply:
@urso I tried implementing this suggestion but I'm not sure if I'm heading in the right direction. So I've created a separate draft PR (based on `master`) isolating this change to make it easier to review: https://github.com/elastic/beats/pull/13273.

I'm curious to know if you think that PR is heading in the right direction before I make it fair. Thanks!